remote/remote_test.go

```diff
@@ -0,0 +1,22 @@
+package remote
+
+import (
+	"testing"
+)
+
+func TestCheckIsJSONObject(t *testing.T) {
+	valuesToTestFalse := []string{
+		"\"Valid JSON String that should be excluded\"",
+		"wrong json string"}
+	valuesToTestTrue := []string{"{\"correct\":\"json object\"}", " { \"correct\" : \"json object with extra spacing\" } "}
+	for _, val := range valuesToTestFalse {
+		if isJSONObject(val) {
+			t.Fatalf("isJSONObject() returned true for %s while it should've been false", val)
+		}
+	}
+	for _, val := range valuesToTestTrue {
+		if !isJSONObject(val) {
+			t.Fatalf("isJSONObject() returned false for %s while it should've been true", val)
+		}
+	}
```

Comment:
The tests are great!

I would just suggest using something called table tests: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

I think it would simplify your tests and also make it more "Go"..

Reply:
Nice! Will introduce them right away ;)